### PR TITLE
cluster: support using provided topo file besides editing it while calling edit-config, add show-config command

### PR DESCRIPTION
**Pull request by haiboumich**
### What problem does this PR solve? 

For subcommand `tiup cluster edit-config`, add a flag `--topology-file` to allow the user to provide a new topology file to substitute the original one besides editing it.

### What is changed and how it works?

 - For command `edit-config`, add a flag `--topology-file` to allow the user to provide a new topology file to substitute the original one besides editing it.
 - add `show-config` command

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 1. Build the new `tiup-cluster` binary and use it as the new component in TiUP central machine.
 2. If the flag `--topology-file` is not specified, it will act like the old flow: 
     * Write Topology to a temporary file. 
     * Open file in editor.
     * Check and update Topology.
     * Save meta file.
 3. Otherwise, it will check the existence of the topology file specified by the flag `--topology-file` and will only 'Check and update Topology.' and 'Save meta file.'.
 4. While executing `./tiup-cluster show-config <cluster-name>`, the cluster's config, which is same as the one shown by opening the topology file to edit while executing `./tiup-cluster edit-config <cluster-name>`, will be printed in console.

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
* Support using a new topology file to substitute the original one besides editing it while calling `tiup cluster edit-config`
* Add the feature `show-config` of component `tiup-cluster` to show cluster's configuration
```

### CODE REVIEW

File: pkg/cluster/manager/edit_config.go [hunk 1]
```diff
@@ -69,34 +74,40 @@ func (m *Manager) EditConfig(name string, skipConfirm bool) error {
...
 // 3. Check and update Topology.
 // 4. Save meta file.
-func (m *Manager) editTopo(origTopo spec.Topology, data []byte, skipConfirm bool) (spec.Topology, error) {
-	file, err := os.CreateTemp(os.TempDir(), "*")
-	if err != nil {
-		return nil, perrs.AddStack(err)
-	}
+func (m *Manager) editTopo(origTopo spec.Topology, data []byte, opt EditConfigOptions, skipConfirm bool) (spec.Topology, error) {
+	var name string
+	if len(opt.NewTopoFile) == 0 {
```
Comment (AstroProfundis) [inline pkg/cluster/manager/edit_config.go:84]: I think it's safe to check `if opt.NewTopoFile == ""`

File: pkg/cluster/manager/edit_config.go [hunk 2]
```diff
@@ -107,8 +118,10 @@ func (m *Manager) editTopo(origTopo spec.Topology, data []byte, skipConfirm bool
 	if err != nil {
 		fmt.Print(color.RedString("New topology could not be saved: "))
 		log.Infof("Failed to parse topology file: %v", err)
-		if pass, _ := tui.PromptForConfirmNo("Do you want to continue editing? [Y/n]: "); !pass {
-			return m.editTopo(origTopo, newData, skipConfirm)
+		if len(opt.NewTopoFile) == 0 {
```
Comment (AstroProfundis) [inline pkg/cluster/manager/edit_config.go:121]: ditto

File: pkg/cluster/manager/edit_config.go [hunk 3]
```diff
@@ -118,8 +131,10 @@ func (m *Manager) editTopo(origTopo spec.Topology, data []byte, skipConfirm bool
 	if err := utils.ValidateSpecDiff(origTopo, newTopo); err != nil {
 		fmt.Print(color.RedString("New topology could not be saved: "))
 		log.Errorf("%s", err)
-		if pass, _ := tui.PromptForConfirmNo("Do you want to continue editing? [Y/n]: "); !pass {
-			return m.editTopo(origTopo, newData, skipConfirm)
+		if len(opt.NewTopoFile) == 0 {
```
Comment (AstroProfundis) [inline pkg/cluster/manager/edit_config.go:134]: ditto

File: pkg/cluster/manager/edit_config.go (hunk 1 above)
Comment (haiboumich) [reply pkg/cluster/manager/edit_config.go]: done

File: pkg/cluster/manager/edit_config.go (hunk 2 above)
Comment (haiboumich) [reply pkg/cluster/manager/edit_config.go]: done

File: pkg/cluster/manager/edit_config.go (hunk 3 above)
Comment (haiboumich) [reply pkg/cluster/manager/edit_config.go]: done

File: components/cluster/command/show_config.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 PingCAP, Inc.
```
Comment (AstroProfundis) [inline components/cluster/command/show_config.go:1]: 2021

File: pkg/cluster/manager/show_config.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 PingCAP, Inc.
```
Comment (AstroProfundis) [inline pkg/cluster/manager/show_config.go:1]: 2021

**Review by AstroProfundis — APPROVED** [review]

File: components/cluster/command/show_config.go (hunk 4 above)
Comment (haiboumich) [reply components/cluster/command/show_config.go]: Modified, thanks for the reminder.

File: pkg/cluster/manager/show_config.go (hunk 5 above)
Comment (haiboumich) [reply pkg/cluster/manager/show_config.go]: Modified.

**Pull request merged**